src/Compatibility/Microsoft.DotNet.PackageValidation/PackageValidationLogger.cs

```diff
@@ -1,22 +1,48 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.ValidationSuppression;
 using Microsoft.NET.Build.Tasks;
 
 namespace Microsoft.DotNet.PackageValidation
 {
     internal class PackageValidationLogger : IPackageLogger
     {
         private readonly Logger _log;
+        private readonly SuppressionEngine _suppressionEngine;
+        private readonly bool _baselineAllErrors;
 
-        public PackageValidationLogger(Logger log)
+        public PackageValidationLogger(Logger log, string baselineFile)
+            : this(log, baselineFile, false) {}
+
+        public PackageValidationLogger(Logger log, string baselineFile, bool baselineAllErrors)
         {
             _log = log;
+            _suppressionEngine = SuppressionEngine.CreateFromSuppressionFile(baselineFile);
```

Comment:
Ideally, `APICompat` should accept a `SuppressionEngine` that way no matter the consumer of `APICompat` always gets the differences baselined?  but that can be done later once `APICompat` has information about the left and right targets.

Reply:
yeah I think we can have a model where both the Logger and ApiCompat take in a suppressionEngine, and they both suppress at different levels. I also agree that this should be done in a separate PR.